executor/runtime/types/types.go

```diff
@@ -491,14 +491,16 @@ func (n *NetworkConfigurationDetails) ToAnnotationMap() map[string]string {
 // pickPrimaryIP applies a series of heuristics to pick the "best" IP that should describe the task.
 // This IP is what ends up show in spinnaker or what gets sshd to, among other things.
 func pickPrimaryIP(n *NetworkConfigurationDetails) string {
+	// In transition mode, we don't want to "leak" the v4 as the primary ip
+	if n.NetworkMode == titus.NetworkConfiguration_Ipv6AndIpv4Fallback.String() {
+		return ""
+	}
 	// If we have an Elastic IP, we consider that to be the main one
 	if n.ElasticIPAddress != "" {
 		return n.ElasticIPAddress
 	}
 
-	// In transition mode, we don't want to "leak" the v4 as the primary ip
-	// But otherwise, for compatibility reasons, we consider this to be the "main" IP
-	if n.IPAddress != "" && n.NetworkMode != titus.NetworkConfiguration_Ipv6AndIpv4Fallback.String() {
+	if n.IPAddress != "" {
```

Comment:
Do you want this below the EIP block on line 507?

I *think* we want to treat the EIP as "the" primary ip we report for the pod if we have it

Reply:
If we're in transition mode, EIPs wont work.